web/controllers/password_controller.ex

```diff
@@ -0,0 +1,21 @@
+defmodule CodeCorps.PasswordController do
+  use CodeCorps.Web, :controller
+
+  alias CodeCorps.{Emails, Mailer, User, AuthToken}
+
+  @doc"""
+  forgot_password should take an email and generate an AuthToken model and send an email
+  """
+  def forgot_password(conn, %{"email" => email}) do
+    with %User{} = user <- Repo.get_by(User, email: email),
+        { :ok, %AuthToken{} = %{ value: token } } <- AuthToken.changeset(%AuthToken{}, user) |> Repo.insert
+    do
+      Emails.ForgotPasswordEmail.create(user, token) |> Mailer.deliver_now()
+    else
+      nil -> nil
+    end
```

Comment:
I would split the `with` functionality off into something in `lib`, some kind of a service or similar. Seeing Chris McCord talk about this, he doesn't appear to think that complex behavior should lie in the controller, nor really in the model.